metomi/isodatetime/data.py

```diff
@@ -312,11 +312,20 @@ def get_prev(self, timepoint: "TimePoint") -> "TimePoint":
 
     def get_first_after(self, timepoint):
         if self._get_is_in_bounds(timepoint):
-            iterations, seconds_since = divmod(
-                (timepoint - self.start_point).get_seconds(),
-                self.duration.get_seconds())
-            return timepoint + (
-                        self.duration - Duration(seconds=floor(seconds_since)))
+            if self.duration.is_exact():
+                # Since it's exact, we can do maths instead of iterating
+                iterations, seconds_since = divmod(
+                    (timepoint - self.start_point).get_seconds(),
+                    self.duration.get_seconds())
+                return timepoint + (self.duration - Duration(
+                    seconds=floor(seconds_since)))
+            else:
+                # Since duration is inexact, we have to iterate
+                current = self.start_point
+                while current <= timepoint:
+                    print(current)
+                    current = self.get_next(current)
```

Comment:
Two things:
- `.get_next()` can return `None`, so need to avoid type error
- don't print
```suggestion
                while current is not None and current <= timepoint:
                    current = self.get_next(current)
```

Reply:
get_next shouldn't ever return None since we've already checked that we're within the valid range. Probably doesn't hurt to confirm it though.

Reply:
Oh, it can if we're sitting directly on the very last instant. got it